src/cmap/auth/gssapi.ts

```diff
@@ -30,60 +51,161 @@ export class GSSAPI extends AuthProvider {
       mechanismProperties['gssapiServiceName'] ||
       'mongodb';
 
-    const MongoAuthProcess = Kerberos.processes.MongoAuthProcess;
+    function externalCommand(
+      command: object,
+      cb: (
+        err: Error | MongoError | undefined,
+        result: { result: { payload: string; conversationId: any } }
+      ) => void
+    ) {
+      return connection.command('$external.$cmd', command, cb);
+    }
+
+    initialize(
+      {
+        retries: 10,
+        host,
+        port,
+        username,
+        password,
+        serviceName: gssapiServiceName,
+        canonicalizeHostName:
```

Comment:
similar to the comment below about `prepare`, this is just a one-off option which should be used during the prepare step and not passed around.